coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala

```diff
@@ -115,45 +115,60 @@ class ShardKeyRegexPlanner(val dataset: Dataset,
     val onKeysReal = ExtraOnByKeysUtil.getRealOnLabels(logicalPlan, queryConfig.addExtraOnByKeysTimeRanges)
 
     val execPlan = if (logicalPlan.operator.isInstanceOf[SetOperator])
-         SetOperatorExec(qContext, inProcessPlanDispatcher, Seq(lhsExec), Seq(rhsExec), logicalPlan.operator,
-          LogicalPlanUtils.renameLabels(onKeysReal, datasetMetricColumn),
-          LogicalPlanUtils.renameLabels(logicalPlan.ignoring, datasetMetricColumn), datasetMetricColumn)
-      else
-         BinaryJoinExec(qContext, inProcessPlanDispatcher, Seq(lhsExec), Seq(rhsExec), logicalPlan.operator,
-          logicalPlan.cardinality, LogicalPlanUtils.renameLabels(onKeysReal, datasetMetricColumn),
-          LogicalPlanUtils.renameLabels(logicalPlan.ignoring, datasetMetricColumn),
-          LogicalPlanUtils.renameLabels(logicalPlan.include, datasetMetricColumn), datasetMetricColumn)
-     PlanResult(Seq(execPlan))
+      SetOperatorExec(qContext, inProcessPlanDispatcher, Seq(lhsExec), Seq(rhsExec), logicalPlan.operator,
```

Comment:
Can we call our as a FIXME the work to optimize the push-down of predicate in the case where both LHS and RHS land up in the same partition ?

Reply:
Done